TestNetworkVerificationOnNonEmptyDB issue

If you run `go test -run TestNetworkVerificationOnNonEmptyDB` multiple times in `node/`, you can sometimes get 
```
--- FAIL: TestNetworkVerificationOnNonEmptyDB (0.63s)
    --- FAIL: TestNetworkVerificationOnNonEmptyDB/different_network (0.30s)
        node_test.go:90:
                Error Trace:    /Users/daniilankusin/juno/node/node_test.go:90
                Error:          An error is expected but got nil.
                Test:           TestNetworkVerificationOnNonEmptyDB/different_network
FAIL
exit status 1
FAIL    github.com/NethermindEth/juno/node      3.524s
```
This happens because `syncBlocks()` does not ensure that the initial synchronisation is completed before `Run(ctx)` returns.
Since `Run(ctx)` returns immediately, the test can check `Head()` before any blocks are actually synchronised and Head is `nil`, so we skip `VerifyBlockHash` and return no error.

I'm not sure yet if we need to fix `syncBlocks()` or just tests. There is [a quick fix](https://github.com/NethermindEth/juno/pull/2494) for the tests.

--- a/node/node_test.go
+++ b/node/node_test.go
@@ -72,8 +72,13 @@ func TestNetworkVerificationOnNonEmptyDB(t *testing.T) {
 			database, err := pebble.New(dbPath)
 			require.NoError(t, err)
 			chain := blockchain.New(database, &network)
-			syncer := sync.New(chain, adaptfeeder.New(feeder.NewTestClient(t, &network)), log, 0, false, database)
-			ctx, cancel := context.WithTimeout(t.Context(), 250*time.Millisecond)
+			ctx, cancel := context.WithCancel(t.Context())
+			syncer := sync.New(chain, adaptfeeder.New(feeder.NewTestClient(t, &network)), log, 0, false, database).WithListener(&sync.SelectiveListener{OnSyncStepDoneCb: func(op string, _ uint64, _ time.Duration) {
+				// Stop the syncer after we successfully stored block.
+				if op == sync.OpStore {
+					cancel()
+				}
+			}})
 			require.NoError(t, syncer.Run(ctx))
 			cancel()
 			require.NoError(t, database.Close())

